clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -3727,15 +3733,24 @@ public void testOffsetValidationRequestGrouping() {
             assertTrue(expectedPartitions.size() > 0);
             allRequestedPartitions.addAll(expectedPartitions);
 
-            Map<TopicPartition, EpochEndOffset> endOffsets = expectedPartitions.stream().collect(Collectors.toMap(
-                Function.identity(),
-                tp -> new EpochEndOffset(Errors.NONE, 4, 0)
-            ));
+            OffsetForLeaderEpochResponseData data = new OffsetForLeaderEpochResponseData();
+            expectedPartitions.forEach(tp -> {
+                OffsetForLeaderTopicResult topic = data.topics().find(tp.topic());
+                if (topic == null) {
```

Comment:
Not for this patch, but all of this boilerplate we need to build the topic groupings gets annoying. It is such a common case that it might be worth writing a special type that lets the parser construct `Map<TopicPartition, Data>` directly since that is really what the code wants. Alternatively, maybe we could flatten the schemas and introduce compression.

Reply:
That would be nice, indeed! I have been thinking about this as well but did not have the time to tackle this yet. I have opened a JIRA to track this: https://issues.apache.org/jira/browse/KAFKA-10795.